modules/apps/27-interchain-accounts/controller/ibc_module.go

```diff
@@ -163,3 +167,22 @@ func (im IBCModule) OnTimeoutPacket(
 
 	return im.app.OnTimeoutPacket(ctx, packet, relayer)
 }
+
+// SendPacket implements the ICS4 Wrapper interface
+func (im IBCMiddleware) SendPacket(
+	ctx sdk.Context,
+	chanCap *capabilitytypes.Capability,
+	packet exported.PacketI,
+) error {
+	panic("SendPacket not supported for ICA controller module")
```

Comment:
ICA auth `SendPacket` should call this function and then this function will pass the call one layer up.

Reply:
https://github.com/cosmos/ibc-go/pull/1248#discussion_r850519226

Reply:
Why would this occur? ICA auth calls `SendTx`

Reply:
ICA auth is being wrapped by ICA correct? In that case we will want the packet flow to be:

`ICA-Auth -SendPacket-> ICA -SendPacket-> IBC`

The ICA middleware in this case will do nothing but forward the packet one layer up. This is to keep consistency and to make sure ICA-auth packets are not missed by additional middleware that is wrapping the stack

Reply:
> ICA auth is being wrapped by ICA correct?

Only partially. `SendPacket` usage is replaced by [`SendTx`](https://ibc.cosmos.network/main/apps/interchain-accounts/auth-modules.html#sendtx)

ICA controller constructs the packet and thus calls [`SendPacket`](https://github.com/cosmos/ibc-go/blob/a1878038ca40d117defd3688ad1494e9872b2ffb/modules/apps/27-interchain-accounts/controller/keeper/relay.go#L70). ICA Auth shouldn't call `SendPacket`. It could bypass `SendTx` by calling `SendPacket` directly, if we panic, we disable this functionality. I don't understand the purpose of `SendTx` if we do allow bypassing. The spec doesn't indicate the [appropriate functionality](https://github.com/cosmos/ibc/tree/master/spec/app/ics-027-interchain-accounts#sendtx)